### PR TITLE
Remove workaround introduced in Synapse v1.50.0rc1 for Mjolnir compatibility. Breaks compatibility with Mjolnir v1.3.1 and earlier.

### PR DESCRIPTION
**Pull request by reivilibre**
Requires https://github.com/matrix-org/mjolnir/pull/218 to land.

Reversion of https://github.com/matrix-org/synapse/pull/11696.

### CODE REVIEW

**Comment by clokep** [comment]
This seems to have extra commits it in FYI!

**Comment by reivilibre** [comment]
Aye, I think they're just artefacts from the release branch but should go away when the release branch is merged into develop.

**Comment by reivilibre** [comment]
The fix was merged into [Mjolnir v1.3.2](https://github.com/matrix-org/mjolnir/releases/tag/v1.3.2) yesterday. I'd be tempted to put a warning in the next Synapse release, *then* cut it out the release after that (v1.55)?

@callahad Is this a sensible plan?

**Comment by callahad** [comment]
SGTM

**Review by babolivier — APPROVED** [review]

File: changelog.d/11700.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove workaround introduced in Synapse v1.50.0rc1 for Mjolnir compatibility. Breaks compatibility with Mjolnir v1.2.1 and earlier.
```
[the remark this answers is not in the document]
Comment (reivilibre) [reply changelog.d/11700.removal:1]: Added the notes, do you think they're good?

File: docs/upgrade.md [hunk 2]
```diff
@@ -106,6 +106,13 @@ You will need to ensure `synctl` is on your `PATH`.
     automatically, though you might need to activate a virtual environment
     depending on how you installed Synapse.
 
+
+## Compatibility dropped for Mjolnir 1.3.1 and earlier
+
+Synapse v1.55.0 drops support for Mjolnir 1.3.1 and earlier.
+If you use the Mjolnir module to moderate your homeserver, please upgrade Mjolnir to version 1.3.2 or later before upgrading Synapse.
```
Comment (babolivier) [inline docs/upgrade.md:113]: Line-wrap please

**Review by babolivier — APPROVED** [review]
Wording lgtm

**Pull request merged**